src/integration-test/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectControllerIntegrationTest.java

```diff
@@ -0,0 +1,320 @@
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.catalog.rest.controller;
+
+import static com.jayway.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.annotation.DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+
+import org.apache.http.HttpStatus;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ow2.proactive.catalog.rest.Application;
+import org.ow2.proactive.catalog.rest.dto.CatalogObjectMetadata;
+import org.ow2.proactive.catalog.rest.entity.Bucket;
+import org.ow2.proactive.catalog.rest.service.CatalogObjectRevisionService;
+import org.ow2.proactive.catalog.rest.service.CatalogObjectService;
+import org.ow2.proactive.catalog.rest.service.repository.BucketRepository;
+import org.ow2.proactive.catalog.rest.service.repository.CatalogObjectRepository;
+import org.ow2.proactive.catalog.rest.util.IntegrationTestUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.google.common.io.ByteStreams;
+import com.jayway.restassured.response.Response;
+
+
+/**
+ * @author ActiveEon Team
+ */
+@ActiveProfiles("test")
+@DirtiesContext(classMode = AFTER_EACH_TEST_METHOD)
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = { Application.class })
+@WebIntegrationTest(randomPort = true)
+public class CatalogObjectControllerIntegrationTest extends AbstractRestAssuredTest {
+
+    private static final String CATALOG_OBJECTS_RESOURCE = "/buckets/{bucketId}/resources";
+
+    private static final String CATALOG_OBJECT_RESOURCE = "/buckets/{bucketId}/resources/{objectId}";
+
+    private static final String contentType = "application/xml";
+
+    @Autowired
+    private BucketRepository bucketRepository;
+
+    @Autowired
+    private CatalogObjectRepository catalogObjectRepository;
+
+    @Autowired
+    private CatalogObjectRevisionService catalogObjectRevisionService;
+
+    @Autowired
+    private CatalogObjectService catalogObjectService;
+
+    private Bucket bucket;
+
+    private CatalogObjectMetadata catalogObject;
+
+    @Before
+    public void setup() throws IOException {
+        bucket = bucketRepository.save(new Bucket("myBucket", "BucketControllerIntegrationTestUser"));
+        catalogObject = catalogObjectService.createCatalogObject(bucket.getId(),
+                                                                 "workflow",
```

Comment:
In integration test we are testing the most important object kind - workflow.
Os in this case "workflow" is a kind.